lib/helpers/rule-test-harness.js

```diff
@@ -91,15 +103,24 @@ module.exports = function generateRuleTests({
 
     let linter;
 
+    function metaToObject(item) {
+      const fallbackMeta = JSON.parse(JSON.stringify(defaultMeta || {}));
+      if (typeof item !== 'object' || item === null) {
+        return fallbackMeta;
+      }
+      if (typeof item.meta === 'object') {
+        return item.meta;
```

Comment:
This should always return a new "clone", so we don't mutate the original shared `item.meta`

Reply:
This still needs to be addressed.